coco_architecture.md

```diff
@@ -0,0 +1,147 @@
+
+- **Attestation service** -  The primary purpose of the attestation service is to validate the evidence provided by the hardware TEE. 
+  This is the verifier as defined in the RATS architecture.
+- **Key Broker Service (KBS)** -  The KBS is the relying party as defined in the RATS architecture.  Following are its primary function:
+  - Receives evidence from the attester (confidential VM or container) via challenge-response protocol
+  - Relay the evidence to the Attestation Service for verification
+  - Apply appraisal policy for the returned Attestation Results to assess the trustworthiness of the attester
+  - Interact with Key Management Service to retrieve the keys and send it to the attester
+- **Key management service** - A service for securely storing, managing and backing up of cryptographic keys used by applications and users.
+- **Image build service** - A service that is used to build confidential container or VM images for end users
+- **Image registry** - A service that is used to store encrypted and/or signed container and VM images required for confidential computing workloads. 
+  Examples: quay.io, docker hub, registries provided by cloud service providers etc.
+
+On the **Confidential VM/Container side** there is this component (**colored green**)
+
+- **Enclave software stack** - The entity responsible for collecting/sending the evidence, getting the encryption keys 
+  and decrypting/running the application. This includes the attestation agent, image management capabilities etc… 
+  
+## Typical Confidential Container Stack using VM based TEEs (TDX/SEV-SNP/SE)
+
+The following diagram shows how a CC solution is used to provide a confidential container to the end user: 
+
+![CC_TEE_container](https://github.com/confidential-containers/documentation/blob/main/images/CC_TEE_container.jpg)
+
+Following is the workflow when deploying a confidential container (Kubernetes POD):
+
+- CC workload preparation
+  - User builds the container image(s)  (podman build -f …)
+  - User signs/encrypts the container image(s)
+  - User pushes the container image(s) to the image registry (podman push …)
+- Deploying the CC workload in k8s
+  - User deploys the workload (kubectl apply -f <...>)
+  - Kubernetes schedules the workload to target host having the required capability to run confidential containers
+- CC workload execution flow (**red connector** in the diagram)
+  - Confidential containers runtime on the host starts the enclave (VM TEE)
+  - Enclave (agent) performs remote attestation: **steps 1-2 from pic**
+  - Enclave (agent) gets the keys required to verify/decrypt the containers image(s):  **steps 3-4 from the pic**
+  - Enclave (image management) downloads the container image(s) : **step 5 from the pic**
+  - Enclave verifies/decrypts the container image(s) : **step 6 from the pic**
+  - Enclave starts the container workload
```

Comment:
Since our core components include the attestation agent, image-rs, and ocicrypt-rs it would be good to depict them here.